src/server/controller-factory.js

```diff
@@ -16,27 +16,29 @@ import DocumentController from './document-controller.js';
 import RevisionController from './revision-controller.js';
 import PdfJsApiController from './pdfjs-api-controller.js';
 import DashboardController from './dashboard-controller.js';
+import RedactionController from './redaction-controller.js';
 import UserAgentController from './user-agent-controller.js';
 
 const controllerTypes = [
-  StaticController,
+  AmbController,
+  RoomController,
   I18nController,
-  UserAgentController,
-  SettingsController,
+  UserController,
   IndexController,
+  ErrorController,
+  AdminController,
+  BatchController,
   SearchController,
-  UserController,
-  DocumentController,
+  StaticController,
   StorageController,
-  BatchController,
-  ErrorController,
-  RoomController,
-  DashboardController,
+  CommentController,
+  SettingsController,
+  DocumentController,
   RevisionController,
   PdfJsApiController,
-  AdminController,
-  AmbController,
-  CommentController
+  DashboardController,
+  RedactionController,
+  UserAgentController
```

Comment:
Even though I strongly appreciate (and admire) the sentiment behind this reordering, the order, in which the middlewares are registered (and then finally called) matters a lot. E.g. the `I18nController` sets the correct language on the `req` object which other controllers further down the pipeline might then use, the `SettingsController` sets `req.settings` which other middleware later might use. The static controller was first so that static file requests e.g. for a font or a script file do not go through the whole application pipeline. The `ErrorController` should be last (I know it somehow has moved into the middle during the last years :thinking:), as it contains the final final middleware that ever should be called. So even though the app might seem to work just fine with the different order, it might cause some subtle bugs that we don't instantly see.

Reply:
😂 Oh I got carried away here!
Thanks for clarifying all the positions, I will try to also change how they are registered so that it's intuitive in the future.

Reply:
Changed!

Reply:
That is some nice organization :heart: thanks!